modules/tinymce/src/plugins/table/test/ts/browser/DetectHeaderRowTest.ts

```diff
@@ -0,0 +1,179 @@
+import { Assertions, Chain, Log, Pipeline, UiFinder } from '@ephox/agar';
+import { UnitTest } from '@ephox/bedrock-client';
+import { TinyApis, TinyLoader } from '@ephox/mcagar';
+import { Element } from '@ephox/sugar';
+import Editor from 'tinymce/core/api/Editor';
+import { detectHeaderRow, getRowType } from 'tinymce/plugins/table/core/TableSections';
+import Plugin from 'tinymce/plugins/table/Plugin';
+import SilverTheme from 'tinymce/themes/silver/Theme';
+
+UnitTest.asynctest('browser.tinymce.plugins.table.DetectHeaderRowTest', (success, failure) => {
+  Plugin();
+  SilverTheme();
+
+  TinyLoader.setupLight((editor: Editor, onSuccess, onFailure) => {
+    const tinyApis = TinyApis(editor);
+
+    Pipeline.async({}, [
+      Log.stepsAsStep('TINY-6007', 'No header rows', [
+        tinyApis.sSetContent(
+          '<table>' +
+          '<tbody><tr>' +
+          '<td>text</td>' +
+          '</tr></tbody>' +
+          '</table>'
+        ),
+        Chain.asStep(Element.fromDom(editor.getBody()), [
```

Comment:
This pattern is reused a lot here, so could we refactor it out, something like this:
```
const sAssertRow = (selector: string, assertions: (row: Element<HTMLTableRowElement>) => void) => 
  Chain.asStep(Element.fromDom(editor.getBody()), [
    UiFinder.cFindIn(selector),
    Chain.op(assertions)
  ]);
```